content/open-source/wolfi/wolfi-with-dockerfiles.md

```diff
@@ -0,0 +1,131 @@
+---
+title: "Creating Wolfi Images with Dockerfiles"
+type: "article"
+description: "This tutorial demonstrates how to build a Wolfi Python image from scratch, using a Dockerfile workflow."
+date: 2022-12-19T08:49:31+00:00
+lastmod: 2022-12-19T08:49:31+00:00
+draft: false
+images: []
+menu:
+  docs:
+    parent: "wolfi"
+weight: 500
+toc: true
+---
+
+## Introduction
+
+[Wolfi](https://edu.chainguard.dev/open-source/wolfi/overview/) is a minimal open source Linux distribution created specifically for cloud workloads, with emphasis on software supply chain security. Using [apk](https://wiki.alpinelinux.org/wiki/Package_management) for package management, Wolfi differs from Alpine in a few important aspects, most notably the use of glib-c instead of Musl and the fact that Wolfi doesn't have a kernel of its own since it is built to run on containers.
+
+In this article, we'll see how to leverage Wolfi to create safer runtime environments based on containers. To demonstrate Wolfi usage in a Dockerfile workflow (using a Dockerfile to build your image), we'll create an image based on the [wolfi-base](https://github.com/chainguard-images/images/tree/main/images/wolfi-base) image maintained by Chainguard. The goal is to have a final runtime image able to execute a Python command-line script.
+
+### Requirements
+
+You'll need a local development environment with Python in order to test the demo, and Docker to build and run the application.
+
+## Obtaining the Demo App
+We'll use the same demo application from the [Getting Started with the Python Chainguard Image](/chainguard/chainguard-images/reference/python/getting-started-python/) tutorial to demonstrate how to build a Wolfi Python image with a Dockerfile. The application files are available on the [edu-images-demos](https://github.com/chainguard-dev/edu-images-demos) repository. We'll start by cloning that repository in a temporary folder so that we can obtain the relevant application files to run the **second** demo from that tutorial.
```

Comment:
```suggestion
We'll use the same demo application from the [Getting Started with the Python Chainguard Image](/chainguard/chainguard-images/reference/python/getting-started-python/) tutorial to demonstrate how to build a Wolfi Python image with a Dockerfile. The application files are available in the [edu-images-demos](https://github.com/chainguard-dev/edu-images-demos) repository. We'll start by cloning that repository in a temporary folder so that we can obtain the relevant application files to run the **second** demo from that tutorial.
```